Pokedex_ios/View/Controller/SortFilter/SortFilterViewController.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  SortFilterViewController.swift
+//  Pokedex_ios
+//
+//  Created by admin on 7/15/22.
+//
+
+import UIKit
+import Foundation
+
+protocol SortButtonActionDelegateDelegate: AnyObject {
+    func sortButtonAction(pokemons: [PokemonRaw])
+}
+
+class SortFilterViewController: PopupViewController {
+    @IBOutlet weak var contentStackView: UIStackView!
+        @IBOutlet weak var sortLabel: UILabel!
+        @IBOutlet weak var sortDescriptionLabel: UILabel!
+        @IBOutlet weak var smallestNumberFirstButton: UIButton!
+        @IBOutlet weak var heighestNumberFirstButton: UIButton!
+        @IBOutlet weak var aToZButton: UIButton!
+        @IBOutlet weak var zToAButton: UIButton!
+       var sortPokemonListViewModel: SortPokemonListViewModel
+
+       init(sortPokemonListViewModel: SortPokemonListViewModel) {
+           self.sortPokemonListViewModel = sortPokemonListViewModel
+           super.init(nibName: nil, bundle: nil)
+       }
+
+       override func viewDidLoad() {
+           super.viewDidLoad()
+           initViewModel()
+           setupContent()
+       }
+
+       func initViewModel() {
+           sortPokemonListViewModel.closePopup = {[weak self] in
+               self?.animateDismissView()
+           }
+       }
+
+       required init?(coder: NSCoder) {
+           fatalError("init(coder:) has not been implemented")
```

Comment:
remove this, if its empty add this:
        super.init(coder: aDecoder)

Reply:
Removed :)